packages/holocron-dev-server/__tests__/lib/webpack/configs/fragments.spec.js

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2021 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  DefinePlugin, DllPlugin, DllReferencePlugin, EnvironmentPlugin,
+} from 'webpack';
+import path from 'path';
+import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+import {
+  getModulesPath,
+  getPublicModulesUrl,
+} from '../../../../lib/utils';
+import {
+  createBrowserConfigFragment,
+  createDllReferenceConfigFragment,
+  createDllBundleConfigFragment,
+  createHolocronModulesConfigFragment,
+  createBundleAnalyzerConfigFragment,
+  createEnvironmentDefinitionsConfigFragment,
+  createResolverConfigFragment,
+  createWatchOptionsConfigFragment,
+} from '../../../../lib/webpack/configs';
+import { getWebpackVersion } from '../../../../lib/webpack';
+
+jest.mock('../../../../lib/webpack/helpers', () => {
+  const originalModule = jest.requireActual('../../../../lib/webpack/helpers');
+
+  return {
+    ...originalModule,
+    getWebpackVersion: jest.fn(() => 5),
+  };
+});
+const { NODE_ENV } = process.env;
+beforeEach(() => {
+  jest.clearAllMocks();
+  process.env.NODE_ENV = NODE_ENV;
+});
+describe('createDllReferenceConfigFragment', () => {
+  test('returns development config for main externals build', () => {
+    const fragment = createDllReferenceConfigFragment();
+    expect(fragment).toEqual({
+      plugins: [
+        new DllReferencePlugin({
+          context: process.cwd(),
+          manifest: `${process.cwd()}/static/vendor/holocron-externals.dll.json`,
+          name: '__externals__',
+        }),
+      ],
+    });
+  });
+});
+describe('createBrowserConfigFragment', () => {
+  test('that source map is returned in development', () => {
+    process.env.NODE_ENV = 'development';
+    const fragment = createBrowserConfigFragment();
+    expect(fragment).toEqual({
+      devtool: 'eval-cheap-source-map',
+      mode: 'development',
+      target: 'web',
+    });
+  });
+  test('that no source map is present in production', () => {
```

Comment:
Why would it ever be production?

Reply:
at the moment, nope, if we do enable `NODE_ENV=production` this would be something we can support, i'll take this out

Reply:
i've removed this test and the functionality because this doesn't support production builds at the moment